src/commands/deploy.js

```diff
@@ -170,15 +168,27 @@ const runDeploy = async ({
       log('Deploying to draft URL...')
     }
 
+    const draft = !deployToProduction && !alias
```

Comment:
We first create the deploy, then upload edge handlers and finally let `js-client` finalise the deploy and upload files and functions.